### ISSUE TITLE
[Bug]: `jsx-no-constructed-context-values` doesn't detect issues when context is used as provider in react v19

### ISSUE THREAD

**Issue by andrew-pledge-io** [jsx-eslint/eslint-plugin-react#3888]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

In React 19, you can render `<Context>` as a provider instead of `<Context.Provider>`.

However when doing so `jsx-no-constructed-context-values` doesn't detect violations.

```tsx
const MyContext = createContext();

function MyComponent(props) {
  return (<>
    {/* old style - violation detected */ }
    <MyContext.Provider value={{ hello: 'world' }} />
    
    {/* react 19 style - no violation */ }
    <MyContext value={{ hello: 'world' }} />
  </>);
}

```

### eslint-plugin-react version

v7.37.2

### eslint version

v8.57.1

### node version

v18.20.6

**Comment by TildaDares** [comment]
I created a PR here #3910
